lib/streamlit/components/v1/components.py

```diff
@@ -194,7 +194,7 @@ def marshall_element_args():
             if key is None:
                 marshall_element_args()
 
-            widget_value = _get_widget_ui_value(
+            widget_value = register_widget(
```

Comment:
Very uncertain and nitpicky of me, but I'm worried that "register_widget" is a better description of what the function _does_, but not _why_ to call it (vs the old name of get_widget_ui_value, anyways). Reading through these code examples, it's not yet clear what "registering" means or why the caller cares?

Especially as we're using it in two different cases kind of interchangeably, one for the `_get_widget_ui_value` and one for `_set_widget_id`. `register` seems more like the latter but we're doing it for both.

I don't have a better alternative, though, and it might just be my knee-jerk reaction to change, so I think we're okay to proceed with `register_widget` (the more I think about it, the better `register_widget` sounds)

Reply:
Yeah, the issue with `_get_widget_ui_value` was that it internally called `_set_widget_id`. So you were always computing and assigning the widget ID (and also registering the widget with the manager) any time you called `_get_widget_ui_value`.

I think a reasonable alternate name for this function would be `register_widget_and_get_ui_value`, which is wordy but more precise.